README.md

```diff
@@ -309,6 +308,195 @@ nnoremap ? <Cmd>call VSCodeNotify('workbench.action.findInFiles', { 'query': exp
+
+##### Actions
+
+###### `vscode.action(name, opts)`
+
+This function is used to run an action asynchronously.
+
+Parameters:
+
+-   `name` (string): The name of the action, generally a vscode command.
+-   `opts` (table): Optional table of options. All fields in the table are optional.
+    -   `args` (table): Optional arguments for the action.
+    -   `range` (table): Specific range for the action. In visual mode, this parameter is generally not needed. There
+        are three supported formats for defining the range (all values are 0-indexed):
+        -   `[start_line, end_line]`
+        -   `[start_line, start_character, end_line, end_character]`
+        -   `{start = { line = start_line, character = start_character}, end = { line = end_line, character = end_character}}`
+    -   `restore_selection` (boolean): Whether to preserve the current selection. Only valid when `range` is specified.
+        Defaults to `true`.
+    -   `callback`: Optional callback function to handle the action result. The callback function should have the
+        following signature: `function(err: string|nil, ret: any)`. The first argument is the error message, and the
+        second is the result. If no callback is provided, any error message will be shown as a notification in VSCode.
+
+###### `vscode.call(name, opts, timeout)`
+
+This function is used to run an action synchronously.
+
+Parameters:
+
+-   `name` (string): The name of the action, generally a vscode command.
+-   `opts` (table): Optional table of options. All fields in the table are optional.
+    -   `args` (table): Optional arguments for the action.
+    -   `range` (table): Specific range for the action. In visual mode, this parameter is generally not needed. There
+        are three supported formats for defining the range (all values are 0-indexed):
+        -   `[start_line, end_line]`
+        -   `[start_line, start_character, end_line, end_character]`
+        -   `{start = { line = start_line, character = start_character}, end = { line = end_line, character = end_character}}`
+    -   `restore_selection` (boolean): Whether to preserve the current selection. Only valid when `range` is specified.
+        Defaults to `true`.
+-   `timeout` (number): Timeout in milliseconds. The default value is -1, which means there is no timeout.
+
+Returns: the result of the action
+
+###### Examples
+
+Currently, two built-in actions are provided for testing purposes:
+
+1. `_ping` returns `"pong"`
+2. `_wait` waits for the specified milliseconds and then returns `"ok"`
+
+```lua
+do -- Execute _ping asynchronously and print the result
+  vscode.action("_ping", {
+    callback = function(err, res)
+      if err == nil then
+        print(res) -- outputs: pong
+      end
+    end,
+  })
+end
+
+-- Format current document
+vscode.action("editor.action.formatDocument")
+
+do -- Comment the three lines below the cursor
+  local lnum = vim.fn.line(".")
```

Comment:
Careful for 1-indexed lines, would be better to have the example show the vim.fn.line(.) - 1